test/integration_test/basic_test.go

```diff
@@ -308,31 +321,128 @@ func BasicCsiRegression(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		if tc.ShouldSkip() {
+			t.Skip()
+		}
 
-		// Sleep for 20 seconds to let operator start the update process
-		logrus.Debug("Sleeping for 20 seconds...")
-		time.Sleep(20 * time.Second)
+		var err error
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
 
-		logrus.Infof("Validate StorageCluster %s", cluster.Name)
-		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateUpdateTimeout, ci_utils.DefaultValidateUpdateRetryInterval, true, "")
+		// Create and validate StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Validate Stork is enabled by default
+		if !cluster.Spec.Stork.Enabled {
+			require.NoError(t, fmt.Errorf("failed to validate Stork is enabled by default, it should be enabled, but it is set to %v", cluster.Spec.Stork.Enabled))
+		}
+
+		// Validate Webhook controller arg doesn't by default
+		if len(cluster.Spec.Stork.Args["webhook-controller"]) != 0 {
+			require.NoError(t, fmt.Errorf("failed to validate webhook-controller, it shouldn't exist by default, but it is set to %s", cluster.Spec.Stork.Args["webhook-controller"]))
+		}
+
+		// Validate HostNetwork is <nil> by default
+		if cluster.Spec.Stork.HostNetwork != nil {
+			require.NoError(t, fmt.Errorf("failed to validate HostNetwork, it shouldn't exist by default, but it is set to %v", *cluster.Spec.Stork.HostNetwork))
+		}
+
+		logrus.Info("Delete stork pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("stork", cluster.Namespace, 120*time.Second)
+		require.NoError(t, err)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
 		require.NoError(t, err)
 
-		logrus.Info("Enable CSI and validate StorageCluster")
+		logrus.Info("Delete stork-scheduler pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("stork-scheduler", cluster.Namespace, 120*time.Second)
+		require.NoError(t, err)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
+		require.NoError(t, err)
+
+		logrus.Info("Enable Stork webhook-controller and validate StorageCluster")
 		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
 		require.NoError(t, err)
-		cluster.Spec.FeatureGates = map[string]string{"CSI": "true"}
-		cluster, err = ci_utils.UpdateStorageCluster(cluster)
+		// At this point this map should be <nil>
+		if len(cluster.Spec.Stork.Args) == 0 {
+			cluster.Spec.Stork.Args = make(map[string]string)
+		}
+		cluster.Spec.Stork.Args["webhook-controller"] = "true"
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
```

Comment:
These are the places where we can directly call `ValidateStork` instead of the complete ValidateStorageCluster. Especially that we are calling it so many times.

Reply:
Done